torch/csrc/jit/codegen/cuda/executor.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+#include <torch/csrc/jit/codegen/cuda/executor_launch_params.h>
+#include <torch/csrc/jit/codegen/cuda/executor_utils.h>
+#include <torch/csrc/jit/codegen/cuda/fusion.h>
+#include <torch/csrc/jit/codegen/cuda/ir_all_nodes.h>
+#include <torch/csrc/jit/codegen/cuda/ir_cloner.h>
+#include <torch/csrc/jit/codegen/cuda/ir_printer.h>
+#include <torch/csrc/jit/codegen/cuda/kernel.h>
+#include <torch/csrc/jit/codegen/cuda/lower2device.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+
+struct TORCH_CUDA_API CompileOptions {
+  int device = 0;
+};
+
+class TORCH_CUDA_API FusionExecutor {
+ public:
+  FusionExecutor() {}
+  FusionExecutor(CompileOptions _options) : options_(_options) {}
+
+  std::string KernelName() const {
+    std::stringstream ss;
+    ss << "kernel"; // << fusion_id;
+    return ss.str();
+  }
+
+  static std::string Namespace() {
+    return "CudaCodeGen";
+  }
+
+  std::string getKernel();
+
+  void compileFusion(Fusion* fusion);
+
+  LaunchParams computeLaunchParams(const at::ArrayRef<IValue>& aten_inputs);
+
+  std::vector<at::Tensor> runFusion(
+      const at::ArrayRef<IValue> inputs,
+      const std::vector<at::Tensor>& outputs);
+
+  std::vector<at::Tensor> runFusion(const at::ArrayRef<IValue> inputs) {
+    TORCH_INTERNAL_ASSERT(false, "Not implemented yet.");
+  }
+
+ private:
+  // TODO: Make pointer to const, will take some const fixing in codegenerator
+  // to work.
+  Fusion fusion_;
```

Comment:
So we don't mutate the input fusion? Nice! (the comment needs to be updated, there's no pointer in here)